tests/test_main.py

```diff
@@ -1,11 +1,125 @@
-    response = client.get("/")
+
+def test_get_projects(client: TestClient, created_projects: list[ProjectDetails]) -> None:
+    response = client.get("/projects")
+    assert response.status_code == 200
+    data = response.json()
+    assert all("project_id" in project for project in data)
+    assert data[0]["name"] == created_projects[0].name
+    assert data[1]["name"] == created_projects[1].name
+    assert data[0]["description"] == created_projects[0].description
+    assert data[1]["description"] == created_projects[1].description
+
+
+def test_get_project_details(client: TestClient, created_project: ProjectDetails) -> None:
+    response = client.get(f"/project/{created_project.project_id}/info")
+    assert response.status_code == 200
+    data = response.json()
+    assert data["name"] == created_project.name
+    assert data["description"] == created_project.description
+    assert data["project_id"] == str(created_project.project_id)
+
+
+def test_get_project_details_not_found(client: TestClient) -> None:
+    random_uuid = uuid.uuid4()
+    response = client.get(f"/project/{random_uuid}/info")
+    assert response.status_code == 404
+    assert response.json() == {"detail": f"ProjectDetails {random_uuid} not found"}
+
+
+def test_update_project_details(client: TestClient, created_project: ProjectDetails) -> None:
+    initial_response = client.get(f"/project/{created_project.project_id}/info")
+    assert initial_response.status_code == 200
+    initial_data = initial_response.json()
+    assert initial_data["name"] == created_project.name
+    assert initial_data["description"] == created_project.description
+    assert initial_data["project_id"] == str(created_project.project_id)
+
+    updated_data = {"name": "Updated Name", "description": "Updated project description"}
+    response = client.put(f"/project/{created_project.project_id}/info", json=updated_data)
     assert response.status_code == 200
-    assert response.json() == {"message": "Hello World"}
+    data = response.json()
+    assert data["name"] == updated_data["name"]
+    assert data["description"] == updated_data["description"]
+    assert data["project_id"] == str(created_project.project_id)
+
+
+def test_update_project_details_not_found(client: TestClient) -> None:
+    random_uuid = uuid.uuid4()
+    updated_data = {"name": "Updated Name", "description": "Updated project description"}
+    response = client.put(f"/project/{random_uuid}/info", json=updated_data)
+    assert response.status_code == 404
+    assert response.json() == {"detail": f"ProjectDetails {random_uuid} not found"}
+
+
+def test_delete_project(client: TestClient, created_project: ProjectDetails) -> None:
+    response = client.delete(f"/project/{created_project.project_id}")
+    assert response.status_code == 204
+    assert response.content == b""
+    get_response = client.get(f"/project/{created_project.project_id}/info")
+    assert get_response.status_code == 404
```

Comment:
same as previous comment - the problem with using another api in test of other endpoint that error in the first may fail the second, even though second doesn't have any issues. Use functions to get value from list or directly check the list. Try to use functions as low as possible